lib/basket.rb

```diff
@@ -1,44 +1,43 @@
 class Basket
-  attr_reader :product
-  attr_accessor :quantity
+  attr_reader :products
+  
+  ProductNotFound = Class.new(StandardError)
 
   def initialize(products = [])
     @products = products
   end
 
 # later: def add(product, quantity=1)
   def add(product)
-    @products << product
+    products << product
     #add warehouse quantity -= 1 later on
   end
 
-  def delete(id)
-    product_in_basket = @products.find { |prod| prod.id == id }
-    if product_in_basket
-      @products.delete(product_in_basket)
-    else
-      puts "product not found in your basket!"
-    end
+  def delete(product_id)
+    product_in_basket = products.find { |prod| prod.id == product_id }
+    raise ProductNotFound unless product_in_basket
+    products.delete(product_in_basket)
   end
 
   def sum
-    @products.
-    map(&:price).
-    reduce(0, :+)    
+    products
+      .map(&:price)
+      .reduce(0, :+)
+      .round(2)    
   end
 
   def sum_with_vat
     (sum * 1.23).round(2)
   end
 
   def show 
-    @products.each do |prod|
-    puts "Product: #{prod.name}, ID: #{prod.id}, price: #{prod.price} \t"
+    products.each do |prod|
+      puts "Product: #{prod.name}, ID: #{prod.id}, price: #{prod.price} \t"
     end
     puts "----------------------------"
     print "Total SUM: " 
     puts sum
     print "With VAT: " 
     puts sum_with_vat
   end
```

Comment:
For me it looks weird that there is a method on model that is responsible for presenting data to the end user. I would put that logic somewhere else if needed.